### PR TITLE
Add transaction gas fee defaults

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## Explanation

Default all the gas fee parameters (`maxFeePerGas`, `maxPriorityFeePerGas`, `gasPrice`) on all new transactions created via the `addTransaction` method in the `TransactionController`.

Generate suggested gas fee values using the `GasFeeController` and a new `getGasFeeEstimates` constructor option.

In addition:

- Add the `userFeeLevel`, `defaultGasEstimates`, and `simulationFails` properties to the transaction metadata.
- Remove the `estimateGasError` property.
- Move all gas and gas fee logic into separate files for modularity and maintainability.
- Create the `helpers` and `utils` directories to organise supporting files.

### `@metamask/transaction-controller`

- **BREAKING**: Remove the `estimateGasError` property from the transaction parameters and `estimateGas` result.
- **CHANGED**: Set default values for all gas fee parameters using the `getGasFeeEstimates` constructor option.
- **ADDED**: Populate the `simulationFails`, `userFeeLevel`, and `defaultGasEstimates` properties in the transaction metadata.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/transaction-controller/src/utils/gas-fees.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+    if (gasEstimateType === GAS_ESTIMATE_TYPES.ETH_GASPRICE) {
+      // The ETH_GASPRICE type just includes a single gas price property,
+      // which we can assume was retrieved from eth_gasPrice
+      return {
+        gasPrice: gweiDecimalToWeiHex(gasFeeEstimates.gasPrice),
+      };
+    }
+  } catch (e) {
+    console.error(e);
```
Comment (vinistevam) [inline packages/transaction-controller/src/utils/gas-fees.ts:265]: Is there a specific reason for not utilizing the `log` function here?

File: packages/transaction-controller/src/types.ts [hunk 2]
```diff
@@ -44,10 +44,13 @@ type TransactionMetaBase = {
    */
   dappSuggestedGasFees?: DappSuggestedGasFees;
 
-  /**
-   * The default estimate for gas.
-   */
-  defaultGasEstimates?: string;
+  defaultGasEstimates?: {
```
Comment (vinistevam) [inline packages/transaction-controller/src/types.ts:47]: Is it beneficial to introduce a distinct type definition for `defaultGasEstimates`?

**Review by OGPoyraz — DISMISSED** [review]

File: packages/transaction-controller/src/utils/gas-fees.ts (hunk 1 above)
Comment (matthewwalsh0) [reply packages/transaction-controller/src/utils/gas-fees.ts]: Done.

File: packages/transaction-controller/src/types.ts (hunk 2 above)
Comment (matthewwalsh0) [reply packages/transaction-controller/src/types.ts]: Done.

**Review by vinistevam — APPROVED** [review]

**Pull request merged**